sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt

```diff
@@ -1,15 +1,19 @@
 package co.nimblehq.sample.compose.ui
 
 import androidx.navigation.*
+import co.nimblehq.sample.compose.model.UiModel
 
 const val KeyId = "id"
+const val ModelKey = "model"
 
 sealed class AppDestination(val route: String = "") {
 
     open val arguments: List<NamedNavArgument> = emptyList()
 
     open var destination: String = route
 
+    open var parcel : Pair<String,UiModel?> = "" to null
```

Comment:
I would recommend having more generic type to support different UiModel. Any please reformat btw 😄 

```suggestion
    open var parcel: Pair<String, Any?> = "" to null
```

I am also thinking about refactoring the `arguments`, seems we don't need to use it anymore, and confusing to have it here. cc @luongvo

Reply:
We need it if we use `navArgument` instead of `savedStateHandle` to pass the data. @hoangnguyen92dn

Reply:
@kaungkhantsoe I see, I mean, could we change to use only one approach for passing data? It would avoid confusion which one they could use case by case from the team.

Reply:
IMO, I would like to leave them as options since using `navArgument` is recommended by the Android Doc. We can also use it if we don't have to pass an object. 💭

Reply:
@kaungkhantsoe If so, could we add the comment to explain when we could use `argument` and `parcel` to clarify the newcomer's confusion? cc @luongvo @Tuubz Please share your though since we will apply this approach for our template. I am thinking about the issue that when `previousBackstackEntry` could be popped out, so we could not receive the data on the destination. Ex: Login -> Home, the Login screen will be popped from the back-stack on logging in successfully.

And I think it better if we reorder the variables a bit: 

```
    open var destination: String = route

    open val arguments: List<NamedNavArgument> = emptyList()

    open var parcel : Pair<String,UiModel?> = "" to null
```

Reply:
@hoangnguyen92dn You mean like this? Is it good enough? 
<img width="1003" alt="Screenshot 2023-01-31 at 12 42 11 PM" src="https://user-images.githubusercontent.com/32578035/215675673-dc6f4a8c-ef75-4eaa-b6ef-29ff62942d09.png">

Reply:
@kaungkhantsoe Looks good to me. One minor comment about the `@param`, I think we can get rid of it. Refer: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments